### ISSUE TITLE
slot missing

### ISSUE THREAD

**Issue by smallollams** [phpredis/phpredis#2360]
I am reading the source code as below:

` int cluster_dist_add_key(redisCluster *c, HashTable *ht, char *key,size_t key_len, clusterKeyVal **kv){
    int key_free;
    short slot;
    clusterDistList *dl;
    clusterKeyVal *retptr;

    // Prefix our key and hash it
    key_free = redis_key_prefix(c->flags, &key, &key_len);
    slot = cluster_hash_key(key, key_len);

    // We can't do this if we don't fully understand the keyspace
    if (c->master[slot] == NULL) {
        if (key_free) efree(key);
        return FAILURE;
    }
...` 

my question is what bad things could lead to the situation “don't fully understand the keyspace”
thank you for reply

**Comment by michael-grunder** [comment]
I don't think it should happen in normal conditions, but this is just defensive programming.

It's always good practice to test for such things in C :smile:

**Comment by smallollams** [comment]
> 

thank you,but i  have encountered a problem. when invoke the "set" method to save data in redis,but finally，my data is not in redis without an exception

**Comment by michael-grunder** [comment]
I'd need more information to help diagnose the issue you're having.

Can you provide a small example that replicates the problem?  The only other suggestion I'd have is to try and do normal debugging to narrow down exactly when this issue is occuring.
